src/components/VoucherModal/index.tsx

```diff
@@ -1,25 +1,42 @@
 import React, { useState, useEffect, useLayoutEffect } from "react"
 import { Skeleton, SkeletonText } from "@chakra-ui/react"
 
+import {
+  redeemVoucher
+} from "api/voucher"
+
 import usePrevious from "hooks/usePrevious"
 import { dateToString } from "utils/date"
 
-import { Modal, ModalProps, Heading, Text } from "@commitUI/index"
+import { Modal, ModalProps, Heading, Text, Button } from "@commitUI/index"
 
 import styles from "./VoucherModal.module.css"
 
 type Props = Omit<ModalProps, "children"> & {
   voucher?: Voucher
   isValidating: boolean
+  user?: User
```

Comment:
Instead of passing user as a prop from the outside, I think it would be better to just do `const { data: user } = useUser();` inside this component. 

Since we are using `SWR`, there is no need to worry about making "duplicate" calls, as `SWR` handles caching for us out of the box, hence even though we have the same API call in many places, it would use the cached value if previous calls has been made :)